### PR TITLE
Implemented paddle.tensor.manipulation.unsqueeze

### LINKED ISSUE #21171: unsqueeze

### PR DESCRIPTION
**Pull request by lspinheiro**
Close #21171

### CODE REVIEW

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_manipulation.py [hunk 1]
```diff
@@ -599,3 +599,43 @@ def test_paddle_roll(
...
+    return draw(st.sampled_from(valid_axes))
+
+
+@handle_frontend_test(
+    fn_tree="paddle.unsqueeze",
+    dtype_param_and_indices=_unsqueeze_helper(),
+)
+def test_paddle_unsqueeze(
+    *,
+    dtype_and_x,
```
Comment (zaeemansari70) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_manipulation.py:625]: You are using `_unsqueeze_helper()` in the decorator, and are not passing in the value in the function below.
You are using `dtype_and_x`which is not there in the decorator.

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
Hi @lspinheiro,

I've highlighted the changes below, kindly request a review once the tests are passing, if you need to ask a question feel free to comment on the PR :+1: 

Thanks!

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_manipulation.py (hunk 1 above)
Comment (lspinheiro) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_manipulation.py]: Hi @zaeemansari70 , thanks for the feedback. Took me a while to understand the test fixtures and I was not able to run the tests the first time. It seems to be working now, let me know if there are any other issues.

```sh
root ➜ /workspaces/ivy (paddle_unsqueeze) $ pytest ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_manipulation.py::test_paddle_unsqueeze
Use Database in ReadOnly Mode with local caching !
=========================================================================================== test session starts ============================================================================================
platform linux -- Python 3.10.0, pytest-7.4.0, pluggy-1.2.0
backend(s): all
device: cpu
number of Hypothesis examples: 25
rootdir: /workspaces/ivy
plugins: anyio-3.7.1, metadata-3.0.0, hypothesis-6.82.0, json-report-1.5.0
collected 5 items                                                                                                                                                                                          

ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_manipulation.py .....  
```

**Comment by lspinheiro** [comment]
> Hi @lspinheiro,
> 
> I've highlighted the changes below, kindly request a review once the tests are passing, if you need to ask a question feel free to comment on the PR :+1: 
> 
> Thanks!

Hi @zaeemansari70 , would you be able to review the PR again after the changes?

Thanks!

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
Your test seem to be failing for all frameworks for a falsifying example which I just provided, kindly please fix this:
```
E           ValueError: (InvalidArgument) The output tensor's rank should be less than 6.
E             [Hint: Expected output_rank <= 6, but received output_rank:7 > 6:6.] (at /paddle/paddle/phi/kernels/funcs/unsqueeze.h:117)
E           
E           Falsifying example: test_paddle_unsqueeze(
E               on_device='cpu',
E               frontend='paddle',
E               backend_fw='jax',
E               dtype_x_and_axis=(['bool'], [array([[[[[[False]]]]]])], 0),
E               fn_tree='ivy.functional.frontends.paddle.unsqueeze',
E               test_flags=FrontendFunctionTestFlags(
E                   num_positional_args=0,
E                   with_out=False,
E                   inplace=False,
E                   as_variable=[False],
E                   native_arrays=[False],
E                   generate_frontend_arrays=False,
E               ),
E           )
E           
E           You can reproduce this example by temporarily adding @reproduce_failure('6.82.0', b'AXicY2JkYEBDcAAAAMkACQ==') as a decorator on your test case
```
Thanks!

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by zaeemansari70** [comment]
Closing due to inactivity

**Pull request closed without merging**